### PR TITLE
Add virtio packed test cases in rng and memballoon devices

### PR DESCRIPTION
**Pull request by meinaLi**
Add virtio packed test cases in rng and memballoon devices

Signed-off-by: Meina Li <meili@redhat.com>

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.setmem.memballoon_option
JOB ID     : 77154ce0444a9dd623266eb099d39e753b9e1eff
JOB LOG    : /root/avocado/job-results/job-2020-12-09T22.15-77154ce/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virsh.setmem.memballoon_option.none_model: PASS (59.76 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.setmem.memballoon_option.remove_driver: PASS (75.35 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.setmem.memballoon_option.with_packed_on: PASS (128.84 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 271.63 s

# avocado run --vt-type libvirt --vt-machine-type q35 libvirt_rng.backend_rdm
JOB ID     : 0b279f473bee4e26d345f38888bb1c5d40e41e54
JOB LOG    : /root/avocado/job-results/job-2020-12-09T22.04-0b279f4/job.log
 (1/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.default.hotplug_unplug.positive.no_options: PASS (36.37 s)
 (2/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.default.hotplug_unplug.positive.persistent: PASS (45.21 s)
 (3/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.default.device_assign: PASS (38.47 s)
 (4/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.detach_device_alias.current.device_assign: PASS (44.96 s)
 (5/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.detach_device_alias.live.device_assign: PASS (44.26 s)
 (6/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.detach_device_alias.config.device_assign: PASS (43.89 s)
 (7/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_rdm.with_packed_on.device_assign: PASS (38.32 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 306.97 s

```

**Comment by meinaLi** [comment]
This PR depends on https://github.com/avocado-framework/avocado-vt/pull/2857

**Review by chunfuwen — APPROVED** [review]
LGTM

File: libvirt/tests/cfg/libvirt_rng.cfg [hunk 1]
```diff
@@ -43,6 +43,10 @@
                             rng_detach_alias_options = "--live"
                         - config:
                             rng_detach_alias_options = "--config"
+                - with_packed_on:
+                  only device_assign
+                  with_packed = "yes"
+                  driver_packed = "on"
```
Comment (kylazhang) [inline libvirt/tests/cfg/libvirt_rng.cfg:49]: The indent seems have some problem

File: libvirt/tests/src/libvirt_rng.py [hunk 2]
```diff
@@ -201,6 +203,8 @@ def check_qemu_cmd(dparams):
         backend_model = dparams.get("backend_model")
         backend_source_list = dparams.get("backend_source",
                                           "").split()
+        with_packed = "yes" == dparams.get("with_packed", "no")
+        driver_packed = dparams.get("driver_packed", "on")
```
Comment (kylazhang) [inline libvirt/tests/src/libvirt_rng.py:207]: seems no need to re-get the params here

File: libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py [hunk 3]
```diff
@@ -266,10 +273,18 @@ def print_debug_stats(original_inside_mem, original_outside_mem,
...
         vmxml.add_device(memballoon_xml)
         logging.info(memballoon_xml)
         vmxml.sync()
         vm.start()
+    # Check guest dumpxml and qemu command in with_packed attribute
+    if with_packed:
+        expect_xml_line = "<driver packed=\"%s\"" % driver_packed
+        libvirt.check_dumpxml(vm, expect_xml_line)
+        expect_qemu_line = "-device virtio-balloon-pci.*packed=%s" % driver_packed
+        libvirt.check_qemu_cmd_line(expect_qemu_line)
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py:287]: I suggest to move the expect xml/qemu line to cfg file for better further update.

File: libvirt/tests/cfg/libvirt_rng.cfg (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/cfg/libvirt_rng.cfg]: Updated

File: libvirt/tests/src/libvirt_rng.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/libvirt_rng.py]: Updated

File: libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py (hunk 3 above)
Comment (meinaLi) [reply libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py:289]: Has moved to the cfg file

**Review by kylazhang — APPROVED** [review]

**Pull request merged**